irods_testing_environment/irods_setup.py

```diff
@@ -350,8 +350,16 @@ def restart_rsyslog(container):
         if ec != 0:
             logging.info(f'[{container.name}] failed to kill rsyslogd')
 
-        ec = execute.execute_command(container, rsyslog_bin_path)
-        if ec != 0:
+        num_attempts = 0
+        while True:
+            ec = execute.execute_command(container, rsyslog_bin_path)
+            if ec != 0:
+                num_attempts += 1
+            is_alive = execute.execute_command(container, f'pgrep {os.path.basename(rsyslog_bin_path)}')
+            if ec == 0 or num_attempts >= 3 or is_alive == 0:
```

Comment:
For some reason, I'm finding `is_alive` is a little confusing here because a return code of 0 actually indicates that `pgrep` found something (which is what we want). I realize it's being used just like an error code with meaning ascribed to it, but it's throwing me off. Personal problem. :)

Maybe we can do the `== 0` on line 358 to make `is_alive` a boolean?

Reply:
Done.